api/nakadi-event-bus-api.yaml

```diff
@@ -723,39 +736,142 @@ definitions:
+    type: object
+    description: |
+      Metadata for this Event.
+
+      Contains commons fields for both Business and DataChange Events. Most are enriched by Nakadi upon
+      reception, but they in general MIGHT be set by the client.
     properties:
-      event_type:
-        type: string
-        example: 'https://resource-events.zalando.com/ResourceCreated'
-      partitioning_key:
-        type: string
-        example: 'ABC123XXX-001'
       metadata:
-        $ref: '#/definitions/EventMetaData'
-  EventMetaData:
+        type: object
+        properties:
+          eid:
+            description: |
+              Identifier of this Event.
+
+              Clients are allowed to generate this and this SHOULD be guaranteed to be unique from the
+              perspective of the producer. Consumers MIGHT use this value to assert uniqueness of reception
+              of the Event.
+
+              If not present upon reception, Nakadi will generate and fill this with a unique value.
+              (TBD: enforce UUID formatif set by the client?)
+            type: string
+            format: uuid
+            example: '105a76d8-db49-4144-ace7-e683e8f4ba46'
+          event-type:
+            description: |
+              The EventType of this Event. This is enriched by Nakadi on reception of the Event based on the
+              endpoint where the Producer sent the Event to.
+
+              If provided MUST match the endpoint. Failure to do so will cause rejection of the Event.
+            type: string
+            example: 'pennybags:payment-business-event'
+          ocurred-at:
+            description: |
+              Timestamp of creation of the Event generated by the producer. Presence is optional but strongly
+              recommended for producers.
+            type: string
+            format: date-time
+            #required: false
+          received-at:
+            type: string
+            description: |
+              Timestamp of the reception of the Event by Nakadi. This is enriched upon reception of the
+              Event.
+              If set by the producer Event will be rejected.
+            format: date-time
+          parent-eid:
+            type: string
+            format: uuid
+            description: |
+              Event identifier of the Event that caused the generation of this Event.
+              Set by the procuder.
+            example: '105a76d8-db49-4144-ace7-e683e8f4ba46'
+            #required: false
+          root-eid:
+            type: string
+            format: uuid
+            description: |
+              Event identifier of the root of the tree of Events (from the relation constructed from
+              parent-eid)
+            example: '105a76d8-db49-4144-ace7-e683e8f4ba46'
+            #required: false
+          flow-id:
+            description: |
+              'required, enriched from header,'
+            type: string
+            example: 'JAh6xH4OQhCJ9PutIV_RYw'
+
```

Comment:
Yes iand is described case by case in the properties themselves. 
Defining what is required in json schema will force clients to put them when this is not really the case.